src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.Reflection.cs

```diff
@@ -548,8 +548,27 @@ private static bool HasInitializer(XamlObjectDefinition objectDefinition)
 			return null;
 		}
 
-		private INamedTypeSymbol GetType(string name)
+		private INamedTypeSymbol GetType(string name, XamlObjectDefinition? objectDefinition = null)
 		{
+			while (objectDefinition is not null)
+			{
+				var namespaces = objectDefinition.Namespaces;
+				if (namespaces is { Count: > 0 } && name.IndexOf(':') is int indexOfColon && indexOfColon > 0)
+				{
+					var ns = name.AsSpan().Slice(0, indexOfColon);
+					foreach (var @namespace in namespaces)
+					{
+						if (ns.Equals(@namespace.Prefix.AsSpan(), StringComparison.Ordinal))
+						{
+							name = $"global::{@namespace.Namespace}.{name.Substring(indexOfColon + 1)}";
```

Comment:
This is not gonna work if `@namespace.Namespace` is not a CLR namespace, for example, if it's `http://schemas.microsoft.com/winfx/2006/xaml/presentation`, but that is a niche case since such namespaces are usually declared in top-level.